src/constraints/field.rs

```diff
@@ -11,65 +12,30 @@ use ark_ff::{One, Zero};
 
 use std::{ops::Neg, sync::Arc};
 
+/// Adds two field elements
+pub fn neg<B: Backend>(
+    compiler: &mut CircuitWriter<B>,
+    cvar: &ConstOrCell<B::Field>,
+    span: Span,
+) -> Var<B::Field> {
+    let res = compiler.backend.enforce_neg_constraint(cvar, span);
+    match res {
+        ConstOrCell::Const(cst) => Var::new_constant(cst, span),
+        ConstOrCell::Cell(cvar) => Var::new_var(cvar, span),
+    }
+}
+
 /// Adds two field elements
 pub fn add<B: Backend>(
     compiler: &mut CircuitWriter<B>,
     lhs: &ConstOrCell<B::Field>,
     rhs: &ConstOrCell<B::Field>,
     span: Span,
 ) -> Var<B::Field> {
-    let zero = B::Field::zero();
-    let one = B::Field::one();
-
-    match (lhs, rhs) {
-        // 2 constants
-        (ConstOrCell::Const(lhs), ConstOrCell::Const(rhs)) => Var::new_constant(*lhs + *rhs, span),
```

Comment:
so my feeling is that it would be more elegant to leave the code handling constants here. Perhaps this means that in the backend we would also have: `add_const(const: F, cvar: CellVar<F>)` and `scale(coeff: F, cvar: CellVar<F>)`

this way the handling of pure constants is not done in the backend

Reply:
this one should be fixed